pkg/codegen/nodejs/importer.go

```diff
@@ -45,6 +45,8 @@ type NodePackageInfo struct {
 	ModuleToPackage map[string]string `json:"moduleToPackage,omitempty"`
 	// Toggle compatibility mode for a specified target.
 	Compatibility string `json:"compatibility,omitempty"`
+	// Disable support for unions in output types.
+	DisableUnionOutputTypes bool `json:"disableUnionOutputTypes,omitempty"`
```

Comment:
Curious - Why does this need to be opt-in?  What are the cases where we *do* want to emit union outputs?  And perhaps knowing the answer to that will make clearer why this is a global setting vs. a per-property setting?

Reply:
`@pulumi/kubernetes` supports unions in output types.

In an ideal world, we wouldn't need this option: instead, the schema would contain separate types for the input and output APIs. Unfortunately, I think that option is too far off at the moment, so I opted for a more pragmatic approach.

Reply:
> @pulumi/kubernetes supports unions in output types.

Again mostly curious - do you have an example of this?  I'm curious if it is common and usable - it seems in general it would be quite hard to consume these values if they truly could come back one of several unrelated types.  

I think it is quite possible that the code-generator emits unions, but that the actual values that come back are guaranteed to be one of the kinds - in which case I think we would generally be "safe" making a breaking change.

This doesn't need to get in the way of the changes here though.

Reply:
One example is `targetPort` in the `ServicePort` interface: https://github.com/pulumi/pulumi-kubernetes/blob/master/sdk/nodejs/types/output.ts#L15433-L15441